docs/upgrade/upgrade_to_v0.10.md

```diff
@@ -0,0 +1,127 @@
+# Upgrade to v0.10
+
+We have test upgrading from v0.8 and later.
+
+Table of Contents
+
+1. [Prepare](#Prepare)
+2. [Stop Services and Backup Data](#Stop-Services-and-Backup-Data)
+3. [Destroy Kubernetes Cluster](#Destroy-Kubernetes-Cluster)
+4. [Install Kubernetes Cluster](#Install-Kubernetes-Cluster)
+5. [Run Migration Scripts And Start Services](#Run-Migration-Scripts-And-Start-Services)
+6. [It's Done](#It's-Done)
+
+## Prepare
+
+### Setup a dev-box of v0.10
+
+All the commands in the document excuted in dev-box.
+You will need to prepare a dev-box of v0.10 first.
+Run the fellow command to create one and work in it:
+
+```bash
+# create dev-box
+sudo docker run -itd \
+        -e COLUMNS=$COLUMNS -e LINES=$LINES -e TERM=$TERM \
+        -v /var/run/docker.sock:/var/run/docker.sock \
+        -v /pathConfiguration:/cluster-configuration  \
+        --pid=host \
+        --privileged=true \
+        --net=host \
+        --name=dev-box \
+        docker.io/openpai/dev-box:v0.10.1
+
+# Working in your dev-box
+sudo docker exec -it dev-box /bin/bash
+cd /pai
+```
+
+### Check PAI cluster version
+
+You could check version from the cluster configuration file `service-configuration.yaml`.
+
+It looks like:
+
+```yaml
+cluster:
+...
+  docker-registry:
+    namespace: openpai
+    domain: docker.io
+    tag: v0.8.3 # It's your cluster version
+    ...
+```
+
+### Backup PAI cluster configuration
+
+If you cluster is v0.9 and later, you could fetch the config from the cluster via paictl:
+
+`./paictl.py config pull -o path_to_backup_your_config`.
+
+There should be four files under the `path_to_backup_your_config`:
+
+1. `layout.yaml` (or `cluster-configuration.yaml` in version 0.8/0.9)
+2. `k8s-role-definition.yaml`
+3. `kubernetes-configuration.yaml`
+4. `services-configuration.yaml`.
+
+### Convert and customize cluster configuration
+
+PAI provide a script tools to convert configuration from old style to the v0.10 release.
+
+Usage:
+
+`./deployment/tools/configMigration.py path_to_backup_your_config path_to_output_new_style_config`.
```

Comment:
Can the files be inplace upgraded, so that it's compatible with most commands. And use the convert command to backup old files somewhere.

Reply:
We want it's easily to trace the changes, for example display the changes using `diff`.
For user experience, a 'upgrade' wizard may solve most of the problems.